packages/create-block/lib/init-package-json.js

```diff
@@ -29,6 +29,55 @@ module.exports = async ( {
 	info( '' );
 	info( 'Creating a "package.json" file.' );
 
+	/**
+	 * Helper to determine if we can install this package.
+	 *
+	 * @param {string} packageArg The package to install.
+	 */
+	function checkDependency( packageArg ) {
+		const { type } = npmPackageArg( packageArg );
+		if (
+			! [ 'git', 'tag', 'version', 'range', 'remote' ].includes( type )
+		) {
+			throw new Error(
+				`Provided package type "${ type }" is not supported.`
+			);
+		}
+	}
+
+	const dependencies = {};
+	const devDependencies = {};
+
+	if ( npmDependencies && npmDependencies.length ) {
+		for ( const packageArg of npmDependencies ) {
+			try {
+				checkDependency( packageArg );
+				const parsed = npmPackageArg( packageArg );
+				dependencies[ parsed.name ] = parsed.fetchSpec || 'latest';
```

Comment:
Should it be `saveSpec`?

> The normalized specifier, for saving to package.json files. null for registry dependencies.

```suggestion
				dependencies[ parsed.name ] = parsed.saveSpec || 'latest';
```

That's the only part I'm unsure about, as this was something that `npm` would handle properly before when installing every package individually.

Reply:
In my testing, if you don't provide the specific version or dist-tag, then it will set it to `latest`. That differs from the old behavior.